.github/workflows/build-and-deploy.yml

```
@@ -0,0 +1,64 @@
+name: build and deploy
+
+on:
+  push:
+  pull_request:
```

```suggestion
on:
  push:
    branches:
      - main
```

I think we want to run this only when PR is merged to main

---

Ofc, I forgot to add this line after deployment tests :))